ui/ui-components/components/record/AddSampleRecordForm.tsx

```diff
@@ -0,0 +1,108 @@
+import { useMemo, useState } from "react";
+import { Form } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { useGrouparooModelContext } from "../../contexts/grouparooModel";
+import { ApiHook } from "../../hooks/useApi";
+import { Actions, Models } from "../../utils/apiData";
+import LoadingButton from "../LoadingButton";
+
+const getInputType = (type?: Models.PropertyType["type"]): string => {
+  switch (type) {
+    case "float":
+    case "integer":
+      return "number";
+    default:
+      return "text";
+  }
+};
+
+interface Props {
+  properties: Models.PropertyType[];
+  onSubmitComplete: (record?: Models.GrouparooRecordType) => void;
+  execApi: ApiHook["execApi"];
+}
+
+const AddSampleRecordForm: React.FC<Props> = ({
+  execApi,
+  onSubmitComplete,
+  properties,
+}) => {
+  const model = useGrouparooModelContext();
+  const { handleSubmit, register } = useForm();
+  const [submitting, setSubmitting] = useState(false);
+
+  const propertiesWithPrimaryKey = useMemo(
+    () =>
+      properties.filter((property) => property.unique && property.isPrimaryKey),
+    [properties]
+  );
+
+  const [selectedUniquePropertyValue, setSelectedUniquePropertyValue] =
+    useState(() => {
+      return propertiesWithPrimaryKey?.[0]?.key || "";
+    });
+
+  const selectedUniqueProperty = useMemo(() => {
+    return propertiesWithPrimaryKey?.find(
+      ({ key }) => key === selectedUniquePropertyValue
+    );
+  }, [selectedUniquePropertyValue]);
+
+  const onSelectUniqueProperty: React.ChangeEventHandler<HTMLInputElement> = (
+    event
+  ) => {
+    setSelectedUniquePropertyValue(event.target.value);
+  };
```

Comment:
I think this is a case for useCallback

Reply:
Not necessarily because this function does not really have any prop/hook dependencies.

Reply:
Based on my understanding of `useCallback`, if you pass it an empty array for deps it will get created once and then never again. So it takes up a little more memory, but requires less time during rendering.

I guess the call would be based on how often this component re-renders.